nucliadb/nucliadb/ingest/orm/knowledgebox.py

```diff
@@ -173,7 +176,15 @@ async def create(
             kb_shards.actual = -1
             # B/c with `Shards.similarity`, replaced by `model`
             kb_shards.similarity = semantic_model.similarity_function
+
+            # if this KB uses a matryoshka model, we can choose a different
+            # dimension
+            if len(semantic_model.matryoshka_dimensions) > 0:
+                semantic_model.vector_dimension = choose_matryoshka_dimension(
```

Comment:
@jotare can you confirm the API contract with choosing the dimension? I guess we need to provide a way for users/ learning to choose different dimensions at kb creation no?